presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcConnection.java

```diff
@@ -335,6 +335,41 @@ private void testRole(String roleParameterValue, ClientSelectedRole clientSelect
         assertEquals(listCurrentRoles(connection), currentRoles);
     }
 
+    @Test
+    public void testSessionProperties()
+            throws SQLException
+    {
+        testSessionProperties(
+                "hive.temporary_staging_directory_path:/tmp;execution_policy:phased",
+                ImmutableMap.of(
+                        "hive.temporary_staging_directory_path", "/tmp",
+                        "execution_policy", "phased"),
+                ImmutableSet.of(
+                        "execution_policy|phased|all-at-once",
+                        "hive.temporary_staging_directory_path|/tmp|/tmp/presto-${USER}"));
+    }
+
+    @Test
+    public void testSessionPropertiesWithMetaCharacters()
+            throws SQLException
+    {
+        testSessionProperties(
+                "hive.temporary_staging_directory_path:/%3A%3B",
+                ImmutableMap.of("hive.temporary_staging_directory_path", "/:;"),
+                ImmutableSet.of("hive.temporary_staging_directory_path|/:;|/tmp/presto-${USER}"));
+    }
+
+    private void testSessionProperties(String urlSessionProperties, Map<String, String> expected, Set<String> expectedProperties)
```

Comment:
What's the difference between `expected` and `expectedProperties`?
Can  you name it more meaningfully?

Reply:
can you name them is some meaningful way?